internal/discoverer/discoverer_test.go

```diff
@@ -26,20 +26,33 @@ func TestServiceEncodeWatch(t *testing.T) {
 				},
 				args: nil,
 			},
-			wantMsg: `key: event, value: update
+			wantMsg: []string{`key: event, value: update
 key: service, value: test
 key: entity, value: upstream;1
 key: entity, value: upstream;2
 key: node, value: 127.0.0.1:80
 key: weight, value: 10
 key: node, value: 127.0.0.1:8080
-key: weight, value: 20`,
-		},
+key: weight, value: 20`, `key: event, value: update
+key: service, value: test
+key: entity, value: upstream;2
```

Comment:
Could we sort the kv during EncodeWatch?

Reply:
Yes.